keps/693-multikueue/README.md

```diff
@@ -0,0 +1,234 @@
+# KEP-693: MultiKueue
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+Introduce an new AdmissionCheck (called MultiKueue) with dedicated API
+and controller that will provide multi-cluster capabilities to Kueue.
+
+## Motivation
+Many of Kueue's users are running multiple clusters and would like to
+have a way to easily distribute batch jobs across them to keep all of
+them utilized. Without a global distribution point, some clusters may
+get less jobs they are able to process, while the others get more, leading
+to underutilization and higher costs. 
+
+### Goals
+* Allow Kueue distribute batch jobs across multiple clusters,
+while maintaining the specified quota limits.
+* Provide users with a single entry point through which the jobs
+can be submitted and monitored, just like they were running in
+a single cluster.
+* Be compatible with all Kueue's features (priorities, borrowing, preemptions, etc)
+and most of integrations.
+* Allow to upgrade single cluster Kueue deployments to multicluster without
+much hassle.
+
+### Non-Goals
+* Solve storage problem. It is assumed that the distributed jobs are
+either location-flexible (for a subset of clusters) or are copying the 
+data as a part of the startup process.
+* Synchronize configuration across the clusters. It is expected that the 
+user will create the appropriate objects, roles and permissions
+in the clusters (manually, using gitops or some 3rd-party tooling).
+* Set up authentication between clusters.
+
+## Proposal
+
+Introduce MultiKueue AdmissionCheck, controller and configuration API. 
+
+Establish the need for a designated management cluster.
```

Comment:
Do you intend to support a leader-follower architecture in the future for management cluster?

Reply:
Can you clarify your proposal? leader-follower just seem like alternate names for this same proposal (manager-workers).

Reply:
There are two answers:

If you don't have global quota (only local-cluster specific). You can have multiple management clusters right now, and send jobs to any of them. The final decision about admission belongs to a local cluster, so the management cluster works only like a proxy, that can be easily switched in case of failure. With the initial design some in-flight jobs may be lost but that issue can be relatively easily fixed. 

If do you have limited global quota: The state of MultiKueue environment exists in both worker clusters and management cluster. It could be possible to recover the state of the broken management cluster from the workers on the other, new management cluster. That currently is not in the design or plans, but could be added later.

Reply:
Can you summarize the ideas around "single point of failure" under risks and mitigations?

Reply:
Done.